pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -253,7 +327,7 @@ func (r *SubnetSetReconciler) CollectGarbage(ctx context.Context) {
 	for subnetSetID := range subnetSetIDsToDelete {
 		nsxSubnets := r.SubnetService.ListSubnetCreatedBySubnetSet(subnetSetID)
 		log.Info("SubnetSet garbage collection, cleaning stale Subnets for SubnetSet", "Count", len(nsxSubnets))
-		if _, err := r.deleteSubnets(nsxSubnets); err != nil {
+		if _, err := r.deleteSubnets(nsxSubnets, true); err != nil {
```

Comment:
Here we delete all the subnets under a subnetset if the subnetset CR is not exists. Just to confirm is this desired to also delete the subnetbindingmap here? It seems that you do not delete the subnetbindingmap in subnet GC when subnet CR does not exist.

Reply:
Yes, here to delete the binding maps in SubnetSet Controller for the auto scale-in case, which is not triggered by by SubnetSet CR event. It is a nsx-operator internal behavior. But for Subnet CR, one CR is mapping one NSX VpcSubnet, and there is no internal behavior to auto scale-in/scale-out Subnets. So we don't introduce internal binding deletion behaviors.